Gateway does not use agent version to reverse proxy requests

**Describe the bug**
These errors can be seen in agent and gateway logs:
```
❬agent-tiny     ❭ [INFO]  ⸬werkzeug:97                    ↦ 127.0.0.1 - - [05/Nov/2025 09:59:30] "GET /v5.1.0/permissions HTTP/1.1" 404 -
❬agent-emulator ❭ [INFO]  ⸬werkzeug:97                    ↦ 127.0.0.1 - - [05/Nov/2025 09:59:30] "GET /v5.1.0/permissions HTTP/1.1" 404 -
❬agent-pocket   ❭ [INFO]  ⸬werkzeug:97                    ↦ 127.0.0.1 - - [05/Nov/2025 09:59:30] "GET /v5.1.0/permissions HTTP/1.1" 404 -
❬gateway        ❭ [ERROR] ⸬slurmweb.views.gateway:134     ↦ Unable to retrieve permissions from cluster tiny: 404
❬gateway        ❭ [ERROR] ⸬slurmweb.views.gateway:134     ↦ Unable to retrieve permissions from cluster atlas: 404
❬gateway        ❭ [ERROR] ⸬slurmweb.views.gateway:134     ↦ Unable to retrieve permissions from cluster pocket: 404
```

The gateway does use the agent version to reverse proxy requests, it wrongly use the minimal agent version from the settings.

**To Reproduce**
n/a

**Expected behavior**
The gateway must use the agent version retrieved from this agent.

**Screenshots**
n/a

**Versions**
5.2.0

**Additional context**
n/a

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [unreleased]
+
+#### Fixed
+- gateway: Use agent provided version instead of agent minimal version from
+  settings to reverse proxy the requests (#656).
+
 ## [5.2.0] - 2025-11-03
 
 ### Added

--- a/slurmweb/views/gateway.py
+++ b/slurmweb/views/gateway.py
@@ -203,7 +203,7 @@ def request_agent(
         if with_version:
             url = (
                 f"{current_app.agents[cluster].url}/"
-                f"v{current_app.settings.agents.version}/{query}"
+                f"v{current_app.agents[cluster].version}/{query}"
             )
         else:
             url = f"{current_app.agents[cluster].url}/{query}"

